mautic/mautic
# Jvectormap next

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [y]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://mautic.atlassian.net/browse/TPROD-440



#### Description:

The hard-coded jvectormap library is not maintained anymore and in https://www.npmjs.com/package/jvectormap there is this alert:

> This package has been deprecated. jvectormap is not maintained since Aug 2015. You can use jvectormap-next or jqvmap instead.

So I replaced the hard-coded library with https://www.npmjs.com/package/jvectormap-next loaded via NPM. We run on the latest version 3.1.1.

There was no need to change any Mautic's code. It all worked when I swapped the library.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. The map is used in 2 places and both load the location from different data.
3. Create a landing page and visit it in an incognito browser. This will create an anonymous contact.
4. Open that contact detail page. If it has the latitude and longitude info in the IP address record then you'll see the Map tab and you'll be able to see the spot on the map:
![Screenshot 2023-05-17 at 17 29 59](https://github.com/mautic/mautic/assets/1235442/81c0f11e-3c28-4539-bd67-7fa72b543bc5)
5. Add the map widget to your dashboard
6. Make sure you have a contact with some country
7. The contact should highlight the country on the map:
![Screenshot 2023-05-17 at 17 30 09](https://github.com/mautic/mautic/assets/1235442/859d0635-627e-45c7-80d4-f39d190c2b72)
